railties/lib/rails/generators/test_unit/scaffold/scaffold_generator.rb

```diff
@@ -11,12 +11,19 @@ class ScaffoldGenerator < Base # :nodoc:
       class_option :api, type: :boolean,
                          desc: "Generates API functional tests"
 
+      class_option :system_tests, type: :string,
+                         desc: "Skip system test files"
+
       argument :attributes, type: :array, default: [], banner: "field:type field:type"
 
       def create_test_files
         template_file = options.api? ? "api_functional_test.rb" : "functional_test.rb"
         template template_file,
                  File.join("test/controllers", controller_class_path, "#{controller_file_name}_controller_test.rb")
+
+        unless options.api? || options[:system_tests].nil?
+          template "system_test.rb", File.join("test/system", class_path, "#{file_name.pluralize}_test.rb")
+        end
```

Comment:
I tried this on master and it's not currently supported. `bin/rails g scaffold Tree title:string --skip-test` also doesn't skip any tests (including model, controller, and system tests). As a result I think adding command line options to non-application generate commands is likely a larger task and should be separate from this change.